### DISCUSSION TITLE
Import Berry script from path

### DISCUSSION THREAD

**Discussion by Slyke** [arendst/Tasmota#15865] [Q&A] [answered]
Hello, trying to import a berry script that's on an SD card.

/sd/run.be:
```
import SomeFunc
var someFunc = SomeFunc()

someFunc.printMe("hello")
```

/sd/someFunc.be:
```
class SomeFunc
  def init()
    print("init func")
  end

  def printMe(text)
    print(text)
  end
end
```

But I get the error:
BRY: Exception> 'type_error' - 'nil' value is not callable

I have also tried:
```
import "/sd/SomeFunc" as SomeFunc
```
But same results. I tried with and without the `.be` extension.

**Comment by s-hadinger** [comment, marked as answer]
imports are done from a list of paths in `sys.path()` and you can't specify a directory in the `import statement`

Something you could do is

``` berry
def push_path(p)
  import sys
  var path = sys.path()
  if path.find(p) == nil  # append only if it's not already there
    path.push(p)
  end
end

push_path("sd")
```

**Reply by sfromis** [reply]
Even ignoring the issue of path handling, with the simple case with everything in the flash FS root, the same error results. As I understand it, `import` is meant for modules, which the class here is not packaged as. 

As `load('SomeFunc.be')` works, I'd expect `load('/sd/SomeFunc.be')` to work too, for a non-module function.

**Reply by Slyke** [reply]
Yep, these worked! Thank you!
